sdk/src/androidTest/java/org/coralibre/android/sdk/internal/DeviceInfoTest.java

```diff
@@ -0,0 +1,72 @@
+        assertEquals("device equals to itself", device1, device1);
+    }
+
+    @Test
+    public void testCompareForDifferentManufacturersCaseInsensitive() {
+        DeviceInfo device1 = new DeviceInfo("asus", "P024_1", "P024", 1, 1, ConfidenceLevel.LOW);
+        DeviceInfo device2 = new DeviceInfo("BLU", "C070", "C4 2019", 1, 1, ConfidenceLevel.LOW);
+        assertTrue("device 1 earlier that device 2", 0 > device1.compareTo(device2));
+        assertTrue("device 1 later that device 2", 0 < device2.compareTo(device1));
+        assertEquals("device equals to itself", device1, device1);
+    }
+
+    @Test
+    public void testCompareForDifferentModels() {
+        DeviceInfo device1 = new DeviceInfo("asus", "K019_2", "K019", 1, 1, ConfidenceLevel.LOW);
+        DeviceInfo device2 = new DeviceInfo("asus", "P024_2", "P024", 1, 1, ConfidenceLevel.LOW);
+        assertTrue("device 1 earlier that device 2", 0 > device1.compareTo(device2));
+        assertTrue("device 1 later that device 2", 0 < device2.compareTo(device1));
+        assertEquals("device equals to itself", device1, device1);
+    }
+
+    @Test
+    public void testCompareForDifferentModelsCaseInsensitive() {
+        DeviceInfo device1 = new DeviceInfo("asus", "K019_2", "K019", 1, 1, ConfidenceLevel.LOW);
+        DeviceInfo device2 = new DeviceInfo("asus", "P024_2", "p024", 1, 1, ConfidenceLevel.LOW);
+        assertTrue("device 1 earlier that device 2", 0 > device1.compareTo(device2));
+        assertTrue("device 1 later that device 2", 0 < device2.compareTo(device1));
+        assertEquals("device equals to itself", device1, device1);
+    }
+
+    @Test
+    public void testCompareForDifferentDevices() {
+        DeviceInfo device1 = new DeviceInfo("asus", "P024_1", "P024", 1, 1, ConfidenceLevel.LOW);
+        DeviceInfo device2 = new DeviceInfo("asus", "P024_2", "P024", 1, 1, ConfidenceLevel.LOW);
+        assertTrue("device 1 earlier that device 2", 0 > device1.compareTo(device2));
+        assertTrue("device 1 later that device 2", 0 < device2.compareTo(device1));
+        assertEquals("device equals to itself", device1, device1);
+    }
+
+    @Test
+    public void testCompareForDifferentDevicesCaseInsensitive() {
+        DeviceInfo device1 = new DeviceInfo("asus", "P024_1", "P024", 1, 1, ConfidenceLevel.LOW);
+        DeviceInfo device2 = new DeviceInfo("asus", "p024_2", "P024", 1, 1, ConfidenceLevel.LOW);
+        assertTrue("device 1 earlier that device 2", 0 > device1.compareTo(device2));
+        assertTrue("device 1 later that device 2", 0 < device2.compareTo(device1));
+        assertEquals("device equals to itself", device1, device1);
+    }
+
+    @Test
+    public void testCaseInsensitivity() {
+        DeviceInfo device1 = new DeviceInfo("AsSs", "p024_1", "P024", 1, 1, ConfidenceLevel.LOW);
+        DeviceInfo device2 = new DeviceInfo("asuS", "P024_1", "p024", 1, 1, ConfidenceLevel.LOW);
+        assertEquals("device equals to itself", device1, device1);
```

Comment:
Here you are just comparing `device1` to itself, which does not test case insensitivity obviously ;-)
```suggestion
        assertEquals("device1 is equal to device2", device1, device2);
```

Reply:
done